roles/matrix-nginx-proxy/defaults/main.yml

```diff
@@ -217,9 +217,35 @@ matrix_nginx_proxy_proxy_domain_additional_server_configuration_blocks: []
 # a new SSL certificate could go into effect.
 matrix_nginx_proxy_reload_cron_time_definition: "20 4 */5 * *"
 
-# Specifies which SSL protocols to use when serving all the various vhosts
+# Specifies the SSL configuration that should be used for the SSL protocols and ciphers
+# This is based on the Mozilla Server Side TLS Recommended configurations.
+#
+# The posible values are:
+# - "Modern" - For Modern clients that support TLS 1.3, with no need for backwards compatibility
+# - "Intermediate" - Recommended configuration for a general-purpose server
+# - "Old" - Services accessed by very old clients or libraries, such as Internet Explorer 8 (Windows XP), Java 6, or OpenSSL 0.9.8
+# - "Custom" - For defining your own protocols an ciphers
+#
+# For more information visit:
+# - https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+# - https://ssl-config.mozilla.org/#server=nginx
+matrix_nginx_proxy_ssl_config: "Intermediate"
+
+# Specifies which *SSL protocols* to use when serving all the various vhosts.
+# This option is ignored except you specify "Custom" in "matrix_nginx_proxy_ssl_config"
 matrix_nginx_proxy_ssl_protocols: "TLSv1.2 TLSv1.3"
```

Comment:
I guess you can make this variable always get used and not only get used when it's custom.

I'd propose something like this:

```yaml
matrix_nginx_proxy_ssl_preset: intermediate

matrix_nginx_proxy_ssl_protocols: "{{ matrix_nginx_proxy_ssl_presets[matrix_nginx_proxy_ssl_preset]['protocols'] }}"

matrix_nginx_proxy_ssl_ciphers: "{{ matrix_nginx_proxy_ssl_presets[matrix_nginx_proxy_ssl_preset]['ciphers'] }}"

matrix_nginx_proxy_ssl_presets:
  intermediate:
    protocols: TLSv1.2 TLSv1.3
    ciphers: ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384
  modern:
    protocols: TLSv1.3
    ciphers: ''
  old:
    protocols: TLSv1 TLSv1.1 TLSv1.2 TLSv1.3
    ciphers: ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA256:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA
  custom:
    protocols: "{{ matrix_nginx_proxy_ssl_custom_protocols }}"
    ciphers: "{{ matrix_nginx_proxy_ssl_custom_ciphers }}"

matrix_nginx_proxy_ssl_custom_protocols: ''
matrix_nginx_proxy_ssl_custom_ciphers: ''
```

Then, our configuration can just use `matrix_nginx_proxy_ssl_protocols` and `matrix_nginx_proxy_ssl_ciphers` without caring about the "preset" type, etc.

---------

To use the "custom" preset would be something like this:

```yaml
matrix_nginx_proxy_ssl_preset: custom
matrix_nginx_proxy_ssl_custom_protocols: 'something'
matrix_nginx_proxy_ssl_custom_ciphers: 'something'
```

-----------

Well, **another thing we could do** is to not have a `custom` preset (defined in `matrix_nginx_proxy_ssl_presets`) at all and just make people "go custom" like this:

```yaml
matrix_nginx_proxy_ssl_protocols: 'something'
matrix_nginx_proxy_ssl_ciphers: ''
```

This is probably better even, as it means we are reusing the `matrix_nginx_proxy_ssl_protocols` variable we always had and people who were redefining it would not experience any change at all.

Unless I'm missing something, this seems like the better solution.